src/visGeometry/index.ts

```diff
@@ -1060,21 +1060,43 @@ class VisGeometry {
         }
     }
 
+    /**
+     * Agent color handling
+     * @property this.colorsData is an array of floats, each 4 floats is a color
+     * @param dataColorIndex is an index into the colorsData array, so it is a multiple of 4
+     * @param colorId is an index into the numberOfColors, so it is a number between 0
```

Comment:
yeah this comment is for the whole section of "agent color handling". I do agree it should be its own module (in addition to a lot of other things in visGeometry).  I was consistent in all the following functions when using these two different names for parameters instead of generic things like "index" or "colorIndex" that we had before. I thought it would be clearer here than having the same comments for every function that used one of these

Reply:
It's the `@param` tags that are not helping.  They look like jsdoc tags for autogenerated documentation and you can't tell what they are parameters for.  I think it would be clearer to have each function get its own jsdoc if you want to leave the code here in the middle of this module.